### PR TITLE
Add ExprPlayerProtocolVersion

### PR DESCRIPTION
**Pull request by bilektugrul**
### Description
Added an expression for getting players' protocol version, exclusive to Paper. Supported all the way back to 1.12.
Also changed ExprProtocolVersion syntax. This can be a breaking change and open to discussion.

---
**Target Minecraft Versions:** any
**Requirements:** Paper
**Related Issues:** #1655

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public class ExprPlayerProtocolVersion extends SimplePropertyExpression<Player, Integer> {
+
+	static {
+		if (Skript.classExists("com.destroystokyo.paper.network.NetworkClient")) {
+			register(ExprPlayerProtocolVersion.class, Integer.class, "[the] [client] protocol version", "players");
+		}
+	}
+
+	@Override
+	public @Nullable Integer convert(Player player) {
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java:35]: The `@Nullable` tag goes under `@Override` and above the public method. Everything else looks good to me. We don't have to worry about default expressions here as we're removing an old syntax that used the existing syntax. So it's good as it other than this micro code standard change.

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprProtocolVersion.java [hunk 2]
```diff
@@ -59,7 +59,7 @@
 public class ExprProtocolVersion extends SimpleExpression<Long> {
 
 	static {
-		Skript.registerExpression(ExprProtocolVersion.class, Long.class, ExpressionType.SIMPLE, "[the] [(sent|required|fake)] protocol version [number]");
+		Skript.registerExpression(ExprProtocolVersion.class, Long.class, ExpressionType.SIMPLE, "[the] [server] list [(sent|required|fake)] protocol version [number]");
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprProtocolVersion.java:62]: Is this done to make the difference between the two expressions more clear? If so, you shouldn't make it a breaking change, maybe just the `[server]` part is good enough?

File: src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	static {
+		if (Skript.classExists("com.destroystokyo.paper.network.NetworkClient")) {
+			register(ExprPlayerProtocolVersion.class, Integer.class, "[the] [client] protocol version", "players");
+		}
+	}
+
+	@Override
+	@Nullable
+	public Integer convert(Player player) {
+		return player.getProtocolVersion();
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java:37]: The documentation says that this method returns `-1` when it is unknown, maybe it'd be a better idea for Skript to make it return null?

**Review by TPGamesNL — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprProtocolVersion.java (hunk 2 above)
Comment (bilektugrul) [reply src/main/java/ch/njol/skript/expressions/ExprProtocolVersion.java:62]: Wouldn't that cause a syntax conflict between ExprPlayerProtocolVersion and ExprProtocolVersion?

Comment (TPGamesNL) [reply src/main/java/ch/njol/skript/expressions/ExprProtocolVersion.java:62]: No, ExprPlayerProtocolVersion has the syntax `of %players%` / `%players%'[s]` added, which ExprProtocolVersion doesn't have

File: src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java (hunk 3 above)
Comment (bilektugrul) [reply src/main/java/ch/njol/skript/expressions/ExprPlayerProtocolVersion.java:37]: I agree with you on that.

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**